src/web/app/pages-student/student-course-details-page/student-course-details-page.component.ts

```diff
@@ -115,26 +119,27 @@ export class StudentCourseDetailsPageComponent implements OnInit {
           }
 
           this.studentProfileService.getStudentProfile(student.email, courseId)
-                .subscribe((studentProfile: StudentProfile) => {
-                  const newPhotoUrl: string =
-                    `${environment.backendUrl}/webapi/student/profilePic`
-                    + `?courseid=${courseId}&studentemail=${student.email}`;
-
-                  const newTeammateProfile: StudentProfileWithPicture = {
-                    studentProfile: {
-                      ...studentProfile,
-                      email: student.email,
-                      shortName: student.name,
-                    },
-                    photoUrl : newPhotoUrl,
-                  };
-
-                  this.teammateProfiles.push(newTeammateProfile);
-                });
+            .subscribe((studentProfile: StudentProfile) => {
+              const newPhotoUrl: string =
+                `${environment.backendUrl}/webapi/student/profilePic`
+                + `?courseid=${courseId}&studentemail=${student.email}`;
+
+              const newTeammateProfile: StudentProfileWithPicture = {
+                studentProfile: {
+                  ...studentProfile,
+                  email: student.email,
+                  shortName: student.name,
+                },
+                photoUrl: newPhotoUrl,
+              };
+
+              this.teammateProfiles.push(newTeammateProfile);
+            });
         });
       }, (resp: ErrorMessageOutput) => {
         this.statusMessageService.showErrorToast(resp.error.message);
       });
+    this.teammateProfilesInit = this.teammateProfiles;
```

Comment:
If the purpose of `teammateProfilesInit` is to hold the initial content of the profiles, shouldn't the original content be first pushed into `teammateProfilesInit` and only then subsequently copied to `teammateProfiles`?

Reply:
You want to say when we push in **teammateProfiles**, I must to copy it directly in **teammateProfilesInit** ?

Reply:
That's one alternative option yes, since this method is only run once I would think it achieves the purpose of a deep copy

Reply:
```suggestion
              this.teammateProfiles.push(newTeammateProfile);
              this.teammateProfilesInit.push(newTeammateProfile);
            });
        });
      }, (resp: ErrorMessageOutput) => {
        this.statusMessageService.showErrorToast(resp.error.message);
      });
      
```